### PR TITLE
cli: plugin sub-command should support installing via url

### PR DESCRIPTION
**Pull request by rmarrcode**

### CODE REVIEW

File: sdk/python/touca/cli/_plugin.py [hunk 1]
```diff
@@ -3,9 +3,10 @@
 from argparse import ArgumentParser
 from pathlib import Path
 import sys
-
+import os
```
Comment (ghorbanzade) [inline sdk/python/touca/cli/_plugin.py:6]: It appears this module is not being used.
```suggestion
```

File: sdk/python/touca/cli/_plugin.py [hunk 2]
```diff
@@ -62,17 +63,27 @@ def __init__(self, options: dict):
 
     def _command_list(self):
         for member in user_plugins():
-            print("{}: {}".format(member.name, member.help))
+            print("{}: {}".format(member.name, member.help))
```
Comment (ghorbanzade) [inline sdk/python/touca/cli/_plugin.py:66]:
```suggestion
            print("{}: {}".format(member.name, member.help))
```

File: sdk/python/touca/cli/_plugin.py [hunk 3]
```diff
@@ -62,17 +63,27 @@ def __init__(self, options: dict):
...
+                if plugin_path_dst.exists():
+                    print(f'plugin "{plugin_name}" is already installed', file=sys.stderr)
+                    return False
+                with open(plugin_path_dst, 'wb') as f:
+                    f.write(res.content)
+                return True
+            #request threw error beacause plugin does not exist
+            except:
+                print(f'did not find a plugin with name "{plugin_name}"', file=sys.stderr)
+                return False
```
Comment (ghorbanzade) [inline sdk/python/touca/cli/_plugin.py:86]: I don't think this change very well described to the reader what we're doing and how the logic fits together. It also borrows logic from before/after.

What do you think about simplifying the impl for this whole function to the following?
```python
    def _command_add(self):
        from shutil import copyfile
        from urllib.parse import urlparse

        plugin_arg: str = self.__options.get("name")
        plugin_name = plugin_arg
        if plugin_arg.startswith("https://"):
            plugin_name = Path(urlparse(plugin_arg).path).stem
        plugin_path_dst = Path(find_home_path(), "plugins", plugin_name).with_suffix(
            ".py"
        )
        plugin_path_dst.parent.mkdir(exist_ok=True)
        if plugin_path_dst.exists():
            print(f'plugin "{plugin_name}" is already installed', file=sys.stderr)
            return False
        plugin_path_src = Path.cwd().joinpath(plugin_name).with_suffix(".py")
        if plugin_path_src.exists():
            copyfile(plugin_path_src, plugin_path_dst)
            return True
        res = requests.get(plugin_arg)
        if res.status_code == 200:
            plugin_path_dst.write_bytes(res.content)
            return True
        print(f'did not find a plugin at "{plugin_arg}"', file=sys.stderr)
        return False
```

**Review by ghorbanzade — CHANGES_REQUESTED** [review]
Thanks so much for the change. As I mentioned in Discord, I think we could further simplify and remove branching complexity. lmk what you think about my comments.

**Comment by ghorbanzade** [comment]
Fixes #130 when merged.

File: sdk/python/touca/cli/_plugin.py (hunk 3 above)
Comment (rmarrcode) [reply sdk/python/touca/cli/_plugin.py:86]: I like this implementation more. I say you just merge this version and I take this as a learning experience in writing concise code

Comment (ghorbanzade) [reply sdk/python/touca/cli/_plugin.py:85]: Thank you for your graciousness, Ryan! Will do. Just trying to improve the readability so we can maintain this code easier.

**Review by ghorbanzade — APPROVED** [review]

**Pull request merged**